server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -202,11 +202,20 @@ public enum MetadataVersion {
     // Add new fetch request version for KIP-951
     IBP_3_7_IV4(19, "3.7", "IV4", false),
 
+    // New version for the Kafka 3.8.0 release.
+    IBP_3_8_IV0(20, "3.8", "IV0", false),
+
+    //
+    // NOTE: MetadataVersions after this point are unstable and may be changed.
+    // If users attempt to use an unstable MetadataVersion, they will get an error.
+    // Please move this comment when updating the LATEST_PRODUCTION constant.
+    //
+
     // Add ELR related supports (KIP-966).
-    IBP_3_8_IV0(20, "3.8", "IV0", true),
+    IBP_3_9_IV0(21, "3.9", "IV0", true),
```

Comment:
Could we add a separate MV for ListOffset?

Reply:
OK. I will put that in 3.9-IV0, and we can put ELR in 3.9-IV1.